docs/configure-custom-values-scalardb-analytics-postgresql.md

```diff
@@ -0,0 +1,179 @@
+# Configure a custom values file for ScalarDB Analytics with PostgreSQL
+
+This document explains how to create your custom values file for the ScalarDB Analytics with PostgreSQL chart. For details on the parameters, see the [README](https://github.com/scalar-labs/helm-charts/blob/main/charts/scalardb-analytics-postgresql/README.md) of the ScalarDB Analytics with PostgreSQL chart.
+
+## Required configurations
+
+### Database configurations
+
+You must set `scalardbAnalyticsPostgreSQL.databaseProperties`. Please set `database.properties` to this parameter. For more details on the configurations of ScalarDB, see [ScalarDB Configurations](https://github.com/scalar-labs/scalardb/blob/master/docs/configurations.md).
```

Comment:
Sorry, I'm a bit confused with this. Does it mean to set `scalardbAnalyticsPostgreSQL.databaseProperties` to the content of `database.properties`?

Reply:
@komamitsu  
Thank you for the question!

It means that the parameter `scalardbAnalyticsPostgreSQL.databaseProperties` should include the properties with the same syntax as the `database.properties` of ScalarDB.

For example, if you want to read two tables in two backend databases via ScalarDB Analytics with PostgreSQL, you need to set the configuration for `Multi-storage transactions` in `scalardbAnalyticsPostgreSQL.databaseProperties`.

* Read two tables data via ScalarDB Analytics with PostgreSQL.

  ```console
                                                  +-----------------------------+
                                            +---> | MySQL ("customer" schema)   |
                                            |     +-----------------------------+
  +------------------------------------+    |
  | ScalarDB Analytics with PostgreSQL | ---+
  +------------------------------------+    |
                                            |     +-----------------------------+
                                            +---> | PostgreSQL ("order" schema) |
                                                  +-----------------------------+
  ```

* Custom values file (setting of `scalardbAnalyticsPostgreSQL.databaseProperties`) in the above case.

  ```yaml
  scalardbAnalyticsPostgreSQL:
    databaseProperties: |
      scalar.db.storage=multi-storage
      scalar.db.multi_storage.storages=storage0,storage1
      
      # Storage 0
      scalar.db.multi_storage.storages.storage0.storage=jdbc
      scalar.db.multi_storage.storages.storage0.contact_points=jdbc:mysql://mysql-scalardb.default.svc.cluster.local:3306/
      scalar.db.multi_storage.storages.storage0.username=root
      scalar.db.multi_storage.storages.storage0.password=mysql
      
      # Storage 1
      scalar.db.multi_storage.storages.storage1.storage=jdbc
      scalar.db.multi_storage.storages.storage1.contact_points=jdbc:postgresql://postgresql-scalardb.default.svc.cluster.local:5432/postgres
      scalar.db.multi_storage.storages.storage1.username=postgres
      scalar.db.multi_storage.storages.storage1.password=postgres
      
      scalar.db.multi_storage.namespace_mapping=customer:storage0,order:storage1
      scalar.db.multi_storage.default_storage=storage1

  schemaImporter:
    namespaces:
      - customer
      - order
  ```

Reply:
Thanks for the details. I think we are on the same page. So, I think there might be a room to make the sentence a bit clearer, but I'll leave it to @josh-wong 😅

Reply:
@kota2and3kan I'm still a bit confused since I'm reading it two different ways. I've provided two suggestions below, so please help me understand which one is accurate🙇🏻‍♂️

> [!NOTE]
> 
> Also, I changed `database.properties` to `ScalarDB .properties file` because I think the file name can be whatever the user chooses (for example, `database.properties`, `scalardb.properties`, etc.). Is that OK?

```suggestion
You must set the `scalardbAnalyticsPostgreSQL.databaseProperties` parameter by following the same syntax that you use to configure the ScalarDB .properties file. For details about configurations, see [ScalarDB Configurations](https://github.com/scalar-labs/scalardb/blob/master/docs/configurations.md).
```

Or, if the contents of `scalardbAnalyticsPostgreSQL.databaseProperties` and the ScalarDB .properties file need to match exactly, the following might be better:

```suggestion
You must set the `scalardbAnalyticsPostgreSQL.databaseProperties` parameter to the same configurations that you have set in the ScalarDB .properties file. For details about configurations, see [ScalarDB Configurations](https://github.com/scalar-labs/scalardb/blob/master/docs/configurations.md).
```

Reply:
@josh-wong  
Thank you for your suggestion!

---

> I've provided two suggestions below, so please help me understand which one is accurate🙇🏻‍♂️

We don't need to exactly match the `database.properties` of ScalarDB (basically, ScalarDB Cluster) and ScalarDB Analytics with PostgreSQL. This is because ScalarDB Analytics with PostgreSQL configuration (`database.properties`) requires only database information that users want to read data via ScalarDB Analytics.

For example, users run the OLTP workload via ScalarDB against 5 databases. However, if they want to run the OLAP workload via ScalarDB Analytics with PostgreSQL against only 3 databases, it is enough that the `database.properties` for ScalarDB Analytics includes configurations for only 3 databases.

```
                                         +------+
                                   +---> | DB 1 |
                                   |     +------+
                                   |
                                   |     +------+
                                   +---> | DB 2 | <---+
                                   |     +------+     |
                                   |                  |
+-------------+    +----------+    |     +------+     |    +------------------------------------+      +-------------+
| OLTP client |--->| ScalarDB | ---+---> | DB 3 | <---+--- | ScalarDB Analytics with PostgreSQL | <--- | OLAP client |
+-------------+    +----------+    |     +------+     |    +------------------------------------+      +-------------+
                                   |                  |
                                   |     +------+     |
                                   +---> | DB 4 | <---+
                                   |     +------+
                                   |
                                   |     +------+
                                   +---> | DB 5 |
                                         +------+
```

In this case, the `database.properties` of ScalarDB and ScalarDB Analytics is different. So, the first one looks good to me.

Also, I think it would be better to add a bit of detail to represent the condition `exact matching is not necessary`.

```suggestion
To access databases via ScalarDB Analytics with PostgreSQL, you must set the `scalardbAnalyticsPostgreSQL.databaseProperties` parameter by following the same syntax that you use to configure the ScalarDB .properties file. For details about configurations, see [ScalarDB Configurations](https://github.com/scalar-labs/scalardb/blob/master/docs/configurations.md).
```

What do you think?

---

> Also, I changed `database.properties` to `ScalarDB .properties file` because I think the file name can be whatever the user chooses (for example, `database.properties`, `scalardb.properties`, etc.). Is that OK?

From the technical perspective, you are right, and I think it's OK to use the term `ScalarDB .properties file`. However, I have one concern. In other ScalarDB documents, I think they state `database.properties` or `scalardb.properties`. In other words, it is not consistent with other documents.

So, there is a possibility that users are confused about `Are the ScalarDB .properties file and database.properties different?`. For example, if users first see ScalarDB documents and then see this document, it causes such confusion I think.

Of course, I think mixing  `database.properties` and `scalardb.properties` in several documents is not good... So, we should unify the terming to `ScalarDB .properties file` in the whole documents of ScalarDB.

As a first step of that, I agree with using the term `ScalarDB .properties file` in this document. But, it affects all ScalarDB documents. So, I think we should discuss terming of properties file with other product members first.

Also, it might be better to update this term in another PR (in another task/story) such as `Unify properties file names in all documents`.

What do you think?

Reply:
> In this case, the database.properties of ScalarDB and ScalarDB Analytics is different. So, the first one looks good to me.
> 
> Also, I think it would be better to add a bit of detail to represent the condition exact matching is not necessary.

@kota2and3kan Thank you for confirming!🙇🏻‍♂️  Let's use the following, with the phrase for the properties file (`database.properties`) for consistency across other docs in this PR.

```suggestion
To access databases via ScalarDB Analytics with PostgreSQL, you must set the `scalardbAnalyticsPostgreSQL.databaseProperties` parameter by following the same syntax that you use to configure the `database.properties` file. For details about configurations, see [ScalarDB Configurations](https://github.com/scalar-labs/scalardb/blob/master/docs/configurations.md).
```

> Also, it might be better to update this term in another PR (in another task/story) such as Unify properties file names in all documents.

I agree. I've only recently found out that the user can specify the name of the `.properties`, so I'd like to note that the file name may vary depending on the user's environment in the future.

As you mentioned, let's determine the exact phrase to use in another PR and across docs. Maybe we can do that after we consolidate docs into one repository to ensure consistency💪

Reply:
@josh-wong  
> Let's use the following, with the phrase for the properties file (`database.properties`) for consistency across other docs in this PR.

Looks good. I will apply it.
Thank you for updating your suggestion!

> As you mentioned, let's determine the exact phrase to use in another PR and across docs. Maybe we can do that after we consolidate docs into one repository to ensure consistency💪

Sound good. Let's discuss and update documents after we unify the docs into one repository!